### PR TITLE
DAOS-3128: soak enhancements

### PR DESCRIPTION
**Pull request by mjean308**
Test-tag-hw-large: pr,hw,large soak_smoke

Updates:
1) create separate logs for failed jobs
2) enable dmg for pool create/destroy
3) enable soak to continue on error
4) change time format to allow hours in yaml

Signed-off-by: Maureen Jean <maureen.jean@intel.com>

### CODE REVIEW

**Comment by mjean308** [comment]
PR failed due to DAOS-4295 FIO small with 256B blocksize

File: src/tests/ftest/soak/soak.yaml [hunk 1]
```diff
@@ -65,8 +68,7 @@ container_reserved:
 # test_params - Defines the type of test to run and how long it runs
 #               It also defines how many pools and jobs to create
 #               name:                The name of the Avocado testcase
-#               test_timeout:        The overall timeout in secs; soak12 will be
-#                                    changed to 43200 once it is stable
+#               test_timeout:        The overall timeout in hours
```
Comment (saurabhtandan) [inline src/tests/ftest/soak/soak.yaml:71]: Optional: Is it possible we can be consistent with test_timeout and timeout to both be in same units either secs or hours. Just for convenience sake.

File: src/tests/ftest/soak/soak.yaml [hunk 2]
```diff
@@ -189,8 +193,8 @@ ior_stress:
         - '512K'
         - '64k'
     daos_oclass:
-          - "SX"
-        # - "RP_2GX"
+        # - "SX"
```
Comment (saurabhtandan) [inline src/tests/ftest/soak/soak.yaml:196]: Is this intentional?  If so why not test both SX and RP_2GX?

Comment (mjean308) [reply src/tests/ftest/soak/soak.yaml:196]: replication is not supported in 0.9 so I have them commented out for now.  Soak is mostly manual and you will tweek it as needed.

File: src/tests/ftest/soak/soak.yaml (hunk 1 above)
Comment (mjean308) [reply src/tests/ftest/soak/soak.yaml:71]: I am going to include a change in test.py to accept timeout in hours  something like 48H or 30M or maybe even 2D  for hour, minutes and days

File: src/tests/ftest/util/apricot/apricot/test.py [hunk 3]
```diff
@@ -82,7 +82,12 @@ def __init__(self, *args, **kwargs):
         # all tests should set a timeout and 60 seconds will enforce that
         if not self.timeout:
             self.timeout = 60
-
+        # Allow yaml timeout value to be in hours; format 2H
+        if "H" in self.timeout:
```
[the remark this answers is not in the document]
Comment (mjean308) [reply src/tests/ftest/util/apricot/apricot/test.py:86]: Phil provided some code to allow using timeout = xD(ays)yH(ours)zM(inutes) param
yaml input:
timeout: 3H | 30M | 5D 3H 50M | 2d 10minutes | 2H 1m should all work but  3H 2D 30M | 2M 5H will not work.  must be specified in DHM order.

        if not self.timeout:
            self.timeout = 60
        elif isinstance(self.timeout, str):
            dhm = re.search(
                r"(?:(\\d+)(?:\\s*d[ays]*\\s*)){0,1}" +
                "(?:(\\d+)(?:\\s*h[ours]*\\s*)){0,1}" +
                "(?:(\\d+)(?:\\s*m[inutes]*)){0,1}",
                self.timeout, re.I).groups()
            self.log.info("DHM: %s", dhm)
            self.timeout = 0
            if dhm[0] is not None:
                self.timeout += 24 * 60 * 60 * int(dhm[0])
            if dhm[1] is not None:
                self.timeout += 60 * 60 * int(dhm[1])
            if dhm[2] is not None:
                self.timeout += 60 * int(dhm[2])
        self.log.info("self.timeout: %s", self.timeout)

File: src/tests/ftest/util/apricot/apricot/test.py [hunk 4]
```diff
@@ -77,11 +77,28 @@ class Test(avocadoTest):
     def __init__(self, *args, **kwargs):
         """Initialize a Test object."""
         super(Test, self).__init__(*args, **kwargs)
+
+        # if self.timeout is a string any time value may be entered
+        # but the time speciified must be in this order DHMS
+        # examples   10D2H10M10S or 25M or 2H5S
+        if isinstance(self.timeout, str):
+            self.timeout = self.timeout.lower()
```
[the remark this answers is not in the document]
Comment (mjean308) [reply src/tests/ftest/util/apricot/apricot/test.py]: self.timeout = str(self.timeout).lower()

**Comment by mjean308** [comment]
CI functional failures have been triaged and are due to DAOS-4467; DAOS-4410 and DAOS-4491. Not related to this PR

**Comment by mjean308** [comment]
Tests are failing with known test issues
DAOS-4410 pool evict bad server name
DAOS-4466 ior dfuse not running
DAOS-4525 FIOsmall - missing binary

File: src/tests/ftest/soak/soak.py [hunk 5]
```diff
@@ -828,9 +835,9 @@ def run_soak(self, test_param):
             self.log.info(
                 "<<PASS %s completed in %s >>", self.loop, DDHHMMSS_format(
                     loop_time))
-            # if the time left if less than a loop exit now
-            if end_time - time.time() < loop_time:
-                break
+            # # if the time left if less than a loop exit now
```
Comment (shimizukko) [inline src/tests/ftest/soak/soak.py:838]:
```suggestion
            # # if the time left is less than a loop exit now
```

Comment (mjean308) [reply src/tests/ftest/soak/soak.py:838]: I will make the change in a future PR; thanks for catch that...

**Pull request merged**